src/System.Windows.Forms/src/System/Windows/Forms/MenuStrip.cs

```diff
@@ -171,7 +171,16 @@ protected virtual void OnMenuActivate(EventArgs e)
         {
             if (IsHandleCreated)
             {
-                AccessibilityNotifyClients(AccessibleEvents.SystemMenuStart, -1);
+                if (!TabStop)
+                {
+                    if (IsAccessibilityObjectCreated && !DesignMode)
+                    {
+                        AccessibilityObject.RaiseAutomationEvent(UiaCore.UIA.MenuModeStartEventId);
+                        AccessibilityObject.RaiseAutomationEvent(UiaCore.UIA.MenuOpenedEventId);
+                    }
+                }
+
+                AccessibilityNotifyClients(AccessibleEvents.SystemMenuStart, User32.OBJID.MENU, -1);
```

Comment:
Conditions could be combined:
```suggestion
                if (!TabStop && !DesignMode && IsAccessibilityObjectCreated)
                {
                    AccessibilityObject.RaiseAutomationEvent(UiaCore.UIA.MenuModeStartEventId);
                    AccessibilityObject.RaiseAutomationEvent(UiaCore.UIA.MenuOpenedEventId);
                }

                AccessibilityNotifyClients(AccessibleEvents.SystemMenuStart, User32.OBJID.MENU, -1);
```
Same for `OnMenuDeactivate`.